docker_cli/_utils.py

```diff
@@ -0,0 +1,256 @@
+import io
+import random
+import re
+from shlex import split
+from string import ascii_lowercase, digits
+from subprocess import DEVNULL, CalledProcessError, run
+from threading import Lock
+from typing import Optional, Tuple, Type
+
+from ._image import Image
+from ._shell_cmds import (
+    PackageManager,
+    URLReader,
+    get_package_manager,
+    get_supported_package_managers,
+    get_supported_url_readers,
+    get_url_reader,
+)
+
+
+def universal_tag_prefix() -> str:
+    """
+    Returns a prefix for tags generated by the system.
+
+    Returns
+    -------
+    str
+        The prefix.
+    """
+    return "dcli"
+
+
+def _package_manager_check(image: Image) -> PackageManager:
+    """
+    Returns the package manager present on an image.
+
+    Parameters
+    ----------
+    base : Image
+        The image.
+
+    Returns
+    -------
+    PackageManager
+        The package manager.
+    """
+    package_mgrs = image.check_command_availability(get_supported_package_managers())
+
+    if package_mgrs:
+        package_mgr = get_package_manager(package_mgrs[0])
+    else:
+        ValueError("No recognized package manager found on parent image.")
+
+    return package_mgr
```

Comment:
No need to check for all possible package managers -- we can break out of the loop as soon as we've found the first available package manager.

```suggestion
    for name in get_supported_package_managers():
        if image.has_command(name):
            return get_package_manager(name)
    else:
        ValueError("No recognized package manager found on parent image.")
```

Reply:
Removing this structure from the two functions that call it in `_utils` results in `image.check_command_availability()` not being called anywhere in the code - would it be better to remove it, or is the functionality potentially useful enough that I should keep it around?

Reply:
I've confirmed that I haven't started calling it in any new places in later branches either

Reply:
I don't really have strong feelings about whether to remove it or not. Maybe the best approach would be to remove it for now if it's unused but keep around a branch that contains the code in case we ever want to add it back.